pkg/apis/pipeline/v1alpha1/pipelinerun_validation.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
+
+// Validate pipelinerun
+func (pr *PipelineRun) Validate() *apis.FieldError {
+	if err := validateObjectMetadata(pr.GetObjectMeta()).ViaField("metadata"); err != nil {
+		return err
+	}
+	return pr.Spec.Validate()
+}
+
+// Validate pipelinerun spec
+func (ps *PipelineRunSpec) Validate() *apis.FieldError {
+	if equality.Semantic.DeepEqual(ps, &PipelineRunSpec{}) {
+		return apis.ErrMissingField("spec")
+	}
+	// pipeline reference should be present for pipelinerun
```

Comment:
It might make sense to validate that the pipeline that the pipelinerun is referencing exists here as well.  I think we should probably do this in our taskrun validation too.  I can make an issue

Reply:
#188 covers the taskrun case

Reply:
Verifying referenced objects at runtime is done in controller. I have updated pipelinerun validation to check for pipeline reference.

Reply:
awesome, thanks!